subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/javadoc/JavadocPlugin.java

```diff
@@ -0,0 +1,183 @@
+package org.shipkit.internal.gradle.javadoc;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.Copy;
+import org.gradle.jvm.tasks.Jar;
+import org.shipkit.gradle.configuration.ShipkitConfiguration;
+import org.shipkit.gradle.git.GitCommitTask;
+import org.shipkit.gradle.git.GitPushTask;
+import org.shipkit.internal.gradle.configuration.ShipkitConfigurationPlugin;
+import org.shipkit.internal.gradle.git.GitCommitTaskFactory;
+import org.shipkit.internal.gradle.git.GitPlugin;
+import org.shipkit.internal.gradle.git.GitUrlInfo;
+import org.shipkit.internal.gradle.git.tasks.CloneGitRepositoryTask;
+import org.shipkit.internal.gradle.git.tasks.GitCheckOutTask;
+import org.shipkit.internal.gradle.util.TaskMaker;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.util.Objects.isNull;
+
+/**
+ * TODO
```

Comment:
It would be super easy to review if you documented in the Javadoc how the task works.